scripts/jointLklDM.C

```diff
@@ -147,18 +151,40 @@ void jointLklDM(TString configFileName="$GLIKESYS/rcfiles/jointLklDM.rc",Int_t s
   TString simulationlabel    = (isSimulation?  "MC"    : "Data");
   TString strprocess         = (isDecay?       "Decay" : "Annihilation");
   
-    // annihilation/decay channel string
+  // Decode the channel and save the decoded channels and branching ratios in channelval and brval, respectively.
+  TObjArray* coefficients = channel.Tokenize("+");
+  Int_t nChannels = coefficients->GetEntries();
+  TString* channelval = new TString[nChannels];
+  Double_t* brval = new Double_t[nChannels];
+  // call the function that decodes the coefficients of the channel
+  decode_channel(coefficients,nChannels,channelval,brval);
+
+  // annihilation/decay channel string
   TString  strchannel;
   Double_t minmass = 0;
-  if     (!channel.CompareTo("bb",TString::kIgnoreCase))         {strchannel = "b #bar{b}";         minmass=5;}
```

Comment:
Later, on line 528, channel is used directly to build the dNdEpSignalFileName, but this is not ok because one should use the equivalent with normalized BRs

Reply:
Good catch!